workloads/vm.go

```diff
@@ -87,47 +215,162 @@ func (v VM) GenerateWorkloadFromVM(vm VM) ([]gridtypes.Workload, error) {
 		Description: vm.Description,
 	}
 	workloads = append(workloads, workload)
-	return workloads, nil
+
+	return workloads
 }
 
-func (v VM) Stage(manager deployer.DeploymentManager, NodeId uint32) error {
-	workloadsMap := map[uint32][]gridtypes.Workload{}
-	workloads, err := v.GenerateWorkloadFromVM(v)
-	if err != nil {
-		return err
+// Dictify converts vm data to a map (dict)
+func (vm *VM) Dictify() map[string]interface{} {
+	envVars := make(map[string]interface{})
+	for key, value := range vm.EnvVars {
+		envVars[key] = value
+	}
+	mounts := make([]interface{}, 0)
+	for _, mountPoint := range vm.Mounts {
+		mount := map[string]interface{}{
+			"disk_name": mountPoint.DiskName, "mount_point": mountPoint.MountPoint,
+		}
+		mounts = append(mounts, mount)
 	}
-	workloadsMap[NodeId] = workloads
-	err = manager.SetWorkloads(workloadsMap)
-	return err
+	zlogs := make([]interface{}, 0)
+	for _, zlog := range vm.Zlogs {
+		zlogs = append(zlogs, zlog.Output)
+	}
+	res := make(map[string]interface{})
+	res["name"] = vm.Name
+	res["description"] = vm.Description
+	res["publicip"] = vm.PublicIP
+	res["publicip6"] = vm.PublicIP6
+	res["planetary"] = vm.Planetary
+	res["corex"] = vm.Corex
+	res["flist"] = vm.Flist
+	res["flist_checksum"] = vm.FlistChecksum
+	res["computedip"] = vm.ComputedIP
+	res["computedip6"] = vm.ComputedIP6
+	res["ygg_ip"] = vm.YggIP
+	res["ip"] = vm.IP
+	res["mounts"] = mounts
+	res["cpu"] = vm.CPU
+	res["memory"] = vm.Memory
+	res["rootfs_size"] = vm.RootfsSize
+	res["env_vars"] = envVars
+	res["entrypoint"] = vm.Entrypoint
+	res["zlogs"] = zlogs
+	res["network_name"] = vm.NetworkName
+	return res
 }
 
-func (zlog *Zlog) GenerateWorkload(zmachine string) gridtypes.Workload {
-	url := []byte(zlog.Output)
-	urlHash := md5.Sum([]byte(url))
-	return gridtypes.Workload{
-		Version: 0,
-		Name:    gridtypes.Name(hex.EncodeToString(urlHash[:])),
-		Type:    zos.ZLogsType,
-		Data: gridtypes.MustMarshal(zos.ZLogs{
-			ZMachine: gridtypes.Name(zmachine),
-			Output:   zlog.Output,
-		}),
+// Validate validates a virtual machine data
+func (vm *VM) Validate() error {
+	if vm.CPU < 1 || vm.CPU > 32 {
+		return errors.Wrap(errors.New("Invalid CPU input"), "CPUs must be more than or equal to 1 and less than or equal to 32")
```

Comment:
we can have a static error `invalid input` and we wrap it with the details, also please make sure errors are all lowercase and don't end with punctuation